docs/enhancements/ova-upload.md

```diff
@@ -0,0 +1,169 @@
+and those arbitrary uploads will be used in migration to construct VMs that will run in the cluster. This is an inherently
+risky operation that is acceptable given the risk already inherent in migrating VMs into the cluster.
+
+The provider server endpoints should be protected by auth, as is done for the inventory endpoints, and a configuration 
+option should be provided so that the administrator can disable the OVA upload/delete endpoints.
+
+## Design Details
+
+### Test Plan
+
+Integration tests should verify that the OVA provider server deployment is reconfigured when the OVA provider's
+storage configuration is changed, that providers configured with NFS shares continue to function, that
+the catalogue management endpoints work, and that uploaded OVAs can be imported.
+
+### Upgrade / Downgrade Strategy
+
+OVA providers that rely on NFS shares can continue to be supported until it becomes reasonable
+to deprecate and remove direct consumption of NFS shares. The provider controller shall update
+any existing OVA provider server deployments to ensure they refer to the correct version of
+the image.
+
+An OVA Provider resource that has been created with a reference to a PVC instead of an NFS share
+will not be able to be inventoried or used in a plan if Forklift is downgraded to a prior version. The existing
+version of Forklift does not update the OVA provider server deployment if the provider is changed, so
+downgrading will require any OVA provider server deployments to be removed so that the provider controller
+can recreate them with the correct version of the image.
+
+## Implementation History
+
+* 1/13/2025: Enhancement pull request opened
+
+## Drawbacks
+
+Allowing the end user to provide a PVC of their choice to use for the appliance catalogue rather than requiring an
+NFS share is a purely beneficial improvement in usability and aligns better with Kube/OpenShift patterns.
+Likewise, allowing upload of images into the store is a significant improvement in usability that comes at the minor
+cost of a slightly more complex provider server to handle the file upload. There are no meaningful drawbacks to these
+usability improvements.
+
+## Alternatives
+
+### Inventorying remote appliances
+
+An alternative to storing appliances locally is to store the URLs of appliances, and then use nbdkit to remotely examine
+the appliance so that it can be added to the inventory. Then the remote appliance disks could be imported directly to 
+VM PVCs without the intermediate step of storing the appliance in the cluster. This avoids the long term storage
+overhead of having the full appliance in the provider catalogue, but it has several drawbacks that make it unsuitable.
+
+* The overhead of transferring the appliance image across the network into the cluster is incurred
+  each time a VM is created from the appliance instead of once when it is added to the catalogue.
```

Comment:
Ah multiple ova imports, interesting and good point!
Won't it cause problems with names duplication or some duplications?

Thinking if the Kubevirt could have something like that.
Maybe we would just import onc have a template VM and then clone it.

Reply:
I think it is most likely that users will want to import appliance images more than once, so we need to consider an approach to automatically generate names for VMs imported from an appliance image. KubeVirt has VM clone functionality, but it requires a KubeVirt feature gate to be enabled and requires the storage drivers to support snapshotting.